### PR TITLE
nanocoap: add convenience function for adding path elements

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Add a convenience function to make adding additional sub-path options easier.

The user might not want to construct the entire path at once (it will be split up internally anyway), e.g. when adding additional resource options.

### Testing procedure

Nothing to test

### CODE REVIEW

File: sys/include/net/nanocoap.h [hunk 1]
```diff
@@ -973,6 +973,22 @@ static inline ssize_t coap_opt_add_format(coap_pkt_t *pkt, uint16_t format)
  */
 ssize_t coap_opt_add_opaque(coap_pkt_t *pkt, uint16_t optnum, const uint8_t *val, size_t val_len);
 
+/**
+ * @brief   Adds one or multiple Uri-Query options in the form '/path' into pkt
```
Comment (kb2ma) [inline sys/include/net/nanocoap.h:977]: s/Uri-Query/Uri-Path/

File: sys/net/application_layer/nanocoap/nanocoap.c [hunk 2]
```diff
@@ -854,6 +854,11 @@ ssize_t coap_opt_add_string(coap_pkt_t *pkt, uint16_t optnum, const char *string
     return write_len;
 }
 
+ssize_t coap_opt_add_path(coap_pkt_t *pkt, const char *path)
```
Comment (kb2ma) [inline sys/net/application_layer/nanocoap/nanocoap.c:857]: Why not inline this implementation in the header?

File: sys/include/net/nanocoap.h [hunk 3]
```diff
@@ -973,6 +973,22 @@ static inline ssize_t coap_opt_add_format(coap_pkt_t *pkt, uint16_t format)
...
+ * @param[in,out] pkt         Packet being built
+ * @param[in]     path        Resource (sub)path
+ *
+ * @pre     ((pkt != NULL) && (path != NULL))
+ *
+ * @return        number of bytes written to pkt buffer
+ * @return        <0 on error
+ * @return        -ENOSPC if no available options or pkt full
+ */
+ssize_t coap_opt_add_path(coap_pkt_t *pkt, const char *path);
```
Comment (kb2ma) [inline sys/include/net/nanocoap.h:990]: Please rename to `coap_opt_add_uri_path()` for consistency with other functions. The principle is to use the canonical name for the option in the function name. There also is a Location-Path option. And yes, yes, I know `coap_opt_add_uquery()` violates this rule. I plan to fix that next. ;-)

File: sys/net/application_layer/nanocoap/nanocoap.c (hunk 2 above)
Comment (benpicco) [reply sys/net/application_layer/nanocoap/nanocoap.c]: I let myself influence by the adjacent `coap_opt_add_uquery()` - fixed.

File: sys/include/net/nanocoap.h [hunk 4]
```diff
@@ -973,6 +973,25 @@ static inline ssize_t coap_opt_add_format(coap_pkt_t *pkt, uint16_t format)
...
+ * @param[in,out] pkt         Packet being built
+ * @param[in]     path        Resource (sub)path
+ *
+ * @pre     ((pkt != NULL) && (path != NULL))
+ *
+ * @return        number of bytes written to pkt buffer
+ * @return        <0 on error
+ * @return        -ENOSPC if no available options or pkt full
+ */
+static inline ssize_t coap_opt_add_uri_path(coap_pkt_t *pkt, const char *path)
```
Comment (kb2ma) [inline sys/include/net/nanocoap.h:990]: Please relocate this function below `coap_opt_add_string()`, which it depends on now that it is inline. This relocation also puts the name of the function in alphabetical order, which is helpful with all of the functions in this file.

**Comment by kb2ma** [comment]
Looks good, please squash.

**Comment by kb2ma** [comment]
Looks like a spurious failure; re-kicked Murdock.

**Comment by benpicco** [comment]
Should be fixed by #13814

**Pull request merged**